src/app/common/user.service.spec.js

```diff
@@ -0,0 +1,159 @@
+        userService.user.nutrition.weight = 40;
+        userService.user.nutrition.height = 150;
+        expect(userService.calculateBMI()).toEqual(17.8);
+      });
+      it('Undefined without any input', function() {
+        expect(userService.calculateBMI()).toBeUndefined();
+        userService.user.nutrition.height = 150;
+        expect(userService.calculateBMI()).toBeUndefined();
+        userService.user.nutrition.weight = 40;
+        expect(userService.calculateBMI()).toBeDefined();
+      });
+    });
+
+    describe('calculate BMR', function() {
+      it('is function', function() {
+        expect(userService.calculateBMR).toEqual(jasmine.any(Function));
+      });
+      it('undefined without inputs', function() {
+        expect(userService.calculateBMR()).toBeUndefined();
+      });
+      it('check formula for man', function() {
+        userService.user.nutrition.weight = 40;
+        expect(userService.calculateBMR()).toBeUndefined();
+        userService.user.nutrition.height = 150;
+        expect(userService.calculateBMR()).toBeUndefined();
+        userService.user.age = 25;
+        expect(userService.calculateBMR()).toBeUndefined();
+
+        userService.user.isMale = true;
+        expect(userService.calculateBMR()).toEqual(1198);
+      });
+      it('check formula for woman', function() {
+        userService.user.nutrition.weight = 40;
+        userService.user.nutrition.height = 150;
+        userService.user.age = 25;
+        userService.user.isMale = false;
+
+        expect(userService.calculateBMR()).toEqual(1208);
+      });
+    });
+    describe('sleepDuration', function() {
+      it('is a function', function() {
+        expect(userService.getUser).toEqual(jasmine.any(Function));
+      });
+      it('returns null without properties defined', function() {
+        expect(userService.sleepDuration()).toBeNull();
+        userService.sleepDuration();
+        expect(userService.sleepDuration()).toBeNull();
+        userService.user.sleep.end = 8;
+        userService.user.sleep.start = 1;
+        expect(userService.sleepDuration()).toBeTruthy();
+      });
+
+      //functionality is not implemented yet
+      xit('both inputs before midnight', function() {
+      });
+      xit('inputs before and after midnight', function() {
+      });
+      xit('both inputs after midnight', function() {
+      });
+    });
+
+    describe('getUser()', function() {
+      it('is a function', function() {
+        expect(userService.getUser).toEqual(jasmine.any(Function));
+      });
+      it('has not been called', function() {
+        spyOn(userService, 'getUser');
+        expect(userService.getUser).not.toHaveBeenCalled();
+      });
+      it('triggers calculateBMI and change user.BMI', function() {
+        expect(userService.user.BMI).toBeNull();
+        spyOn(userService, 'calculateBMI').and.callThrough();
+        userService.getUser();
+        expect(userService.calculateBMI).toHaveBeenCalled();
```

Comment:
- [ ] refactor

[Now it's much cleaner.](https://github.com/VasylPosmit/health/pull/17/commits/4a3de23aa1a951807d6161095e90c12fbdb59bd1)
KISS